### PR TITLE
Implemented a colorpicker for all color selectors. #204, #395

### PR DESCRIPTION
**Pull request by ossie**
Added a colorpicker for both boards and labels. 

Signed-off-by: Oskar Kurz <oskar.kurz@gmail.com>

Fix for https://github.com/nextcloud/deck/issues/204 and https://github.com/nextcloud/deck/issues/395

### CODE REVIEW

**Comment by juliushaertl** [comment]
Thank you very much for the pull request. Looks great so far.

Only a small nitpick - the color fields / picker in the left sidebar should all have the same width. 
bildschirmfoto vom 2018-01-22 20-24-48

File: templates/part.boardlist.php [hunk 1]
```diff
@@ -102,12 +105,12 @@ class="ng-pristine ng-valid" ng-submit="boardCreate()">
 					<input class="edit ng-valid ng-empty"
 						   type="text" placeholder="<?php p($l->t('New board title')); ?>"
 						   autofocus-on-insert ng-model="newBoard.title" maxlength="100" ng-model-options="{ debounce: 250 }">
-					<div class="colorselect">
-						<div class="color" ng-repeat="c in ::colors"
-							 ng-style="{'background-color':'#{{ c }}'}"
-							 ng-click="selectColor(c)"
-							 ng-class="{'selected': (c == newBoard.color), 'dark': (newBoard.color | textColorFilter) === '#ffffff' }"></div>
-					</div>
+                    <div class="colorselect" ng-controller="ColorPickerController">
```
Comment (pixelipo) [inline templates/part.boardlist.php]: There seems to be some issues with spacing here. Please fix

**Comment by pixelipo** [comment]
We'll need to move this icon over to server (it's the same one used in Calendar app and probably some other ones as well) and clean it up (it contains a lot of unnecessary code), but that is out of the scope of this PR.

File: templates/part.boardlist.php (hunk 1 above)
Comment (ossie) [reply templates/part.boardlist.php]: I'm gonna look into this.

**Comment by ossie** [comment]
I fixed the colorpicker width.

**Comment by juliushaertl** [comment]
@ossie Thanks for the fix and sorry for the long delay reviewing. Can you check the color selector in the left sidebar, when editing a board? The custom color picker seems to be hidden there now.

**Comment by plyaskin** [comment]
How to refresh this in the Nextcloud?

**Comment by juliushaertl** [comment]
@ossie Would it be ok for you if I take over to fix the last minor bugs?

**Comment by juliushaertl** [comment]
Closing in favour of #468 so I can do the final small adjustments.

Thanks for your hard work on this awesome feature @ossie I really appreciate it.

**Comment by ossie** [comment]
@juliushaertl Sorry for not reacrting at all. The last months were stuffed with other (not programming related) projects. Thanks for fixing the last bugs.

**Pull request closed without merging**